src/env.rs

```diff
@@ -26,6 +26,9 @@ pub struct Environment {
 
     /// Parses booleans, integers and floats if they're detected (can be safely parsed).
     try_parsing: bool,
+
+    /// Alternate source for environment. This can be used for testing.
+    source: Option<Map<String, String>>,
```

Comment:
I'm not sure whether you mean testing of this crate or testing when using this crate.

If the former, this should be `#[cfg(test)]` IMO! Otherwise it would be nice to have an example here,... or some words how it can be used!

Reply:
The latter. I added a few more words and an example. I hope this helps.